From edb82112851f6274eef5fb4dc99be745809814d2 Mon Sep 17 00:00:00 2001
From: Luc Verhaegen <libv@codethink.co.uk>
Date: Tue, 23 Oct 2012 20:05:56 +0200
Subject: [PATCH 875/944] video:sunxi:disp: fix wrong type for right shifting

Found by sparse: warning: right shift by bigger than source value.

Signed-off-by: Luc Verhaegen <libv@codethink.co.uk>
---
 drivers/video/sunxi/disp/de_fe.c    | 12 ++++++------
 drivers/video/sunxi/disp/de_layer.c | 13 +++++++------
 drivers/video/sunxi/disp/ebios_de.h | 38 ++++++++++++++++++-------------------
 3 files changed, 32 insertions(+), 31 deletions(-)

diff --git a/drivers/video/sunxi/disp/de_fe.c b/drivers/video/sunxi/disp/de_fe.c
index 4661d92..60ba759 100644
--- a/drivers/video/sunxi/disp/de_fe.c
+++ b/drivers/video/sunxi/disp/de_fe.c
@@ -522,12 +522,12 @@ __s32 DE_SCAL_Set_Scaling_Factor(__u8 sel, __scal_scan_mod_t * in_scan,
  *   out_type:  output data format
  *   smth_mode: scaler filter effect select
  */
-__s32 DE_SCAL_Set_Scaling_Coef(__u8 sel, __scal_scan_mod_t * in_scan,
-			       __scal_src_size_t * in_size,
-			       __scal_src_type_t * in_type,
-			       __scal_scan_mod_t * out_scan,
-			       __scal_out_size_t * out_size,
-			       __scal_out_type_t * out_type, __u8 smth_mode)
+__s32 DE_SCAL_Set_Scaling_Coef(__u8 sel, __scal_scan_mod_t *in_scan,
+			       __scal_src_size_t *in_size,
+			       __scal_src_type_t *in_type,
+			       __scal_scan_mod_t *out_scan,
+			       __scal_out_size_t *out_size,
+			       __scal_out_type_t *out_type, __u32 smth_mode)
 {
 	__s32 in_w0, in_h0, in_w1, in_h1, out_w0, out_h0, out_w1, out_h1;
 	__s32 ch0h_smth_level, ch0v_smth_level, ch1h_smth_level,
diff --git a/drivers/video/sunxi/disp/de_layer.c b/drivers/video/sunxi/disp/de_layer.c
index 70e0ea7..fb2e593 100644
--- a/drivers/video/sunxi/disp/de_layer.c
+++ b/drivers/video/sunxi/disp/de_layer.c
@@ -108,17 +108,18 @@ __s32 DE_BE_Layer_Set_Work_Mode(__u32 sel, __u8 layidx, __u8 mode)
 	return 0;
 }
 
-static __s32 DE_BE_Layer_Set_Addr(__u32 sel, __u8 layidx, __u32 addr)	//bit
+/*
+ * bit
+ */
+static __s32 DE_BE_Layer_Set_Addr(__u32 sel, __u8 layidx, __u32 addr)
 {
 	__u32 tmp_l, tmp_h, tmp;
 	tmp_l = addr << 3;
-	tmp_h = (__u8) (addr & 0xe0000000) >> 29;
+	tmp_h = (__u8) ((addr & 0xe0000000) >> 29);
 	DE_BE_WUINT32IDX(sel, DE_BE_FRMBUF_LOW32ADDR_OFF, layidx, tmp_l);
 
-	tmp =
-	    DE_BE_RUINT32(sel,
-			  DE_BE_FRMBUF_HIGH4ADDR_OFF) & (~(0xff <<
-							   (layidx * 8)));
+	tmp = DE_BE_RUINT32(sel, DE_BE_FRMBUF_HIGH4ADDR_OFF) &
+		(~(0xff << (layidx * 8)));
 	DE_BE_WUINT32(sel, DE_BE_FRMBUF_HIGH4ADDR_OFF,
 		      tmp | (tmp_h << (layidx * 8)));
 
diff --git a/drivers/video/sunxi/disp/ebios_de.h b/drivers/video/sunxi/disp/ebios_de.h
index e94fd3d..b03ebf5 100644
--- a/drivers/video/sunxi/disp/ebios_de.h
+++ b/drivers/video/sunxi/disp/ebios_de.h
@@ -287,25 +287,25 @@ __u32 DE_SCAL_Get_Reg_Base(__u8 sel);
 __s32 DE_SCAL_Config_Src(__u8 sel, __scal_buf_addr_t * addr,
 			 __scal_src_size_t * size, __scal_src_type_t * type,
 			 __u8 field, __u8 dien);
-__s32 DE_SCAL_Set_Fb_Addr(__u8 sel, __scal_buf_addr_t * addr);
-__s32 DE_SCAL_Set_Init_Phase(__u8 sel, __scal_scan_mod_t * in_scan,
-			     __scal_src_size_t * in_size,
-			     __scal_src_type_t * in_type,
-			     __scal_scan_mod_t * out_scan,
-			     __scal_out_size_t * out_size,
-			     __scal_out_type_t * out_type, __u8 dien);
-__s32 DE_SCAL_Set_Scaling_Factor(__u8 sel, __scal_scan_mod_t * in_scan,
-				 __scal_src_size_t * in_size,
-				 __scal_src_type_t * in_type,
-				 __scal_scan_mod_t * out_scan,
-				 __scal_out_size_t * out_size,
-				 __scal_out_type_t * out_type);
-__s32 DE_SCAL_Set_Scaling_Coef(__u8 sel, __scal_scan_mod_t * in_scan,
-			       __scal_src_size_t * in_size,
-			       __scal_src_type_t * in_type,
-			       __scal_scan_mod_t * out_scan,
-			       __scal_out_size_t * out_size,
-			       __scal_out_type_t * out_type, __u8 smth_mode);
+__s32 DE_SCAL_Set_Fb_Addr(__u8 sel, __scal_buf_addr_t *addr);
+__s32 DE_SCAL_Set_Init_Phase(__u8 sel, __scal_scan_mod_t *in_scan,
+			     __scal_src_size_t *in_size,
+			     __scal_src_type_t *in_type,
+			     __scal_scan_mod_t *out_scan,
+			     __scal_out_size_t *out_size,
+			     __scal_out_type_t *out_type, __u8 dien);
+__s32 DE_SCAL_Set_Scaling_Factor(__u8 sel, __scal_scan_mod_t *in_scan,
+				 __scal_src_size_t *in_size,
+				 __scal_src_type_t *in_type,
+				 __scal_scan_mod_t *out_scan,
+				 __scal_out_size_t *out_size,
+				 __scal_out_type_t *out_type);
+__s32 DE_SCAL_Set_Scaling_Coef(__u8 sel, __scal_scan_mod_t *in_scan,
+			       __scal_src_size_t *in_size,
+			       __scal_src_type_t *in_type,
+			       __scal_scan_mod_t *out_scan,
+			       __scal_out_size_t *out_size,
+			       __scal_out_type_t *out_type, __u32 smth_mode);
 __s32 DE_SCAL_Set_CSC_Coef(__u8 sel, __u8 in_csc_mode, __u8 out_csc_mode,
 			   __u8 incs, __u8 outcs, __u32 in_br_swap,
 			   __u32 out_br_swap);
-- 
1.8.0

